src/compiler/moduleSpecifiers.ts

```diff
@@ -338,11 +365,11 @@ namespace ts.moduleSpecifiers {
                         endsWith(relativeToBaseUrl, suffix) ||
                         !suffix && relativeToBaseUrl === removeTrailingDirectorySeparator(prefix)) {
                         const matchedStar = relativeToBaseUrl.substr(prefix.length, relativeToBaseUrl.length - suffix.length);
-                        return key.replace("*", matchedStar);
+                        return { key, path: key.replace("*", matchedStar) };
                     }
                 }
                 else if (pattern === relativeToBaseUrl || pattern === relativeToBaseUrlWithIndex) {
-                    return key;
+                    return { key, path: key };

```

Comment:
Please rename something so that this doesn't happen

Reply:
"this" being the `{ b, a: b }` form

Reply:
This was unneeded after all, reverted the change